src/swarm/neo/node/IRequestHandler.d

```diff
@@ -64,3 +66,40 @@ public interface IRequestHandler
 
     void postSupportedCodeSent ( );
 }
+
+/// ditto
+public interface IRequest
+{
+    import ocean.transition;
+
+    import swarm.neo.node.RequestOnConn;
+
+    /***************************************************************************
+
+        Passes the request-on-conn and request resource acquirer to the handler.
+
+        Params:
+            connection = request-on-conn in which the request handler is called
+            resources = request resources acquirer
+
+    ***************************************************************************/
+
+    void initialise ( RequestOnConn connection, Object resources );
```

Comment:
Note: I left the separate `initialise` method as it's quite convenient to have a mixin that provides an implementation of this method along with class fields to store the provided references.

Reply:
You can still have this method in your mixin and start your `handle` implementation with `this.initialise(connection, resources);`. It would be one extra line and keep the interface simpler.

Reply:
Ok yeah true, that makes sense. Will update.